### PR TITLE
Add subscription to bank services api #37

### LINKED ISSUE #37: Add subscription to bank services API
The main goal of this feature is to allow a user to subscribe to bank services. 

1. Create a changelog with test bank services:
- name: "account summary", price: $10 (1000c), periodicity: one time
- name: "premium", price: $20, periodicity: weekly
- "credit score", price: $5, periodicity: monthly
3. Create API: POST "bank/services/subscriptions" with request body "SubscriptionRequestDto" with the following fields: long userId, long accountId, long bankServiceId. The successful response should be "SubscriptionResponseDto" with the following fields: long accountId, Subscription subscription.
3. Perform the following checks: 
- account money is enough for payment
- the account belongs to the correct user
4. If something is wrong, then return 400 Bad Request.
5. If everything is correct, create a subscription record and save it in the DB. 
6. Withdraw money (service payment) from the account. 
7. Return "SubscriptionResponseDto".

### PR DESCRIPTION
**Pull request by andy-sam**
fixes #37

### CODE REVIEW

File: src/main/java/com/codesarge/banking/controllers/SubscriptionController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public SubscriptionController(
+            SubscriptionService subscriptionService) {
+        this.subscriptionService = subscriptionService;
+    }
+
+    @PostMapping("bank/services/subscriptions")
+    public ResponseEntity<SubscriptionResponseDto> makeSubscription(
+            @RequestBody SubscriptionRequestDto subscriptionRequestDto) {
+        log.info("Requested creating subscription for user with id " + subscriptionRequestDto
+                .getUserId() + " and account # " + +subscriptionRequestDto.getAccountId());
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/SubscriptionController.java:30]: redundant plus is here. please remove

File: src/main/java/com/codesarge/banking/service/SubscriptionService.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        if (account.getBalance() < bankService.getPrice())
+            return null;
+//        checking if the account belongs to the correct user
+        if (!account.getUser().getId().equals(subscriptionRequestDto.getUserId()))
+            return null;
+
+        Subscription subscription = new Subscription(bankService, account);
+        Subscription savedSubscription = subscriptionRepository.save(subscription);
+        SubscriptionDto savedSubscriptionDto = new SubscriptionDto(savedSubscription);
+//        withdraw money (service payment) from the account
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/SubscriptionService.java:46]: redundant comment. please remove

**Review by alex-anakin — CHANGES_REQUESTED** [review]
Just one more thing. You implemented SubscriptionResponseDto as a wrapper of SubscriptionDto. So it's redundant. Please remove SubscriptionResponseDto at all and return SubscriptionDto instead.

Everything else looks great.

File: src/main/java/com/codesarge/banking/controllers/SubscriptionController.java (hunk 1 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/SubscriptionController.java:30]: done

File: src/main/java/com/codesarge/banking/service/SubscriptionService.java (hunk 2 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/SubscriptionService.java:46]: done

**Comment by andy-sam** [comment]
file:///home/andreew/Pictures/Screenshot%20from%202021-07-18%2000-21-19.png
/home/andreew/Pictures/Screenshot%20from%202021-07-18%2000-24-09.png

**Review by alex-anakin — APPROVED** [review]

**Pull request merged**